### ISSUE TITLE
torch.compile Inductor produces wrong output for InstanceNorm2d + normalize + hardswish chain

### ISSUE THREAD

**Issue by ALinrunrun** [pytorch/pytorch#183601]
### 🐛 Describe the bug

`torch.compile` with the Inductor backend produces incorrect output for a pure PyTorch model containing `InstanceNorm2d`, `AdaptiveAvgPool2d`, `F.normalize`, multiplication, and `F.hardswish`.

Eager execution and compiled execution produce significantly different results. The maximum absolute difference is about `1.9985`.

```
import torch
import torch.nn as nn
import torch.nn.functional as F

torch.manual_seed(42)

x = torch.randn(4, 3, 10, 12)
inst_norm = nn.InstanceNorm2d(3)
pool = nn.AdaptiveAvgPool2d((1, 1))

def m():
    t = inst_norm(x)
    t = pool(t)
    t = F.normalize(t, dim=0)
    t = t * x
    t = F.hardswish(t)
    return F.normalize(t, dim=0)

torch._dynamo.reset()
eager = m()

torch._dynamo.reset()
compiled = torch.compile(m, backend="inductor")()

print("eager vs compile max_diff:", (eager - compiled).abs().max().item())
```

### Error logs

`eager vs compile max_diff: 1.9985179901123047`

### Versions

PyTorch version: 2.11.0+cu130
Is debug build: False
CUDA used to build PyTorch: 13.0
ROCM used to build PyTorch: N/A

OS: Ubuntu 24.04.2 LTS (x86_64)
GCC version: (Ubuntu 13.3.0-6ubuntu2~24.04.1) 13.3.0
Clang version: Could not collect
CMake version: Could not collect
Libc version: glibc-2.39

Python version: 3.10.20 (main, Mar 11 2026, 17:46:40) [GCC 14.3.0] (64-bit runtime)
Python platform: Linux-5.15.167.4-microsoft-standard-WSL2-x86_64-with-glibc2.39
Is CUDA available: False
CUDA runtime version: Could not collect
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: GPU 0: NVIDIA GeForce RTX 3080 Laptop GPU
Nvidia driver version: 545.92
cuDNN version: Could not collect
Is XPU available: False
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True
Caching allocator config: N/A

Versions of relevant libraries:
[pip3] numpy==2.2.6
[pip3] onnx==1.21.0
[pip3] onnx2torch==1.5.15
[pip3] onnxruntime==1.23.2
[pip3] torch==2.11.0
[pip3] torchvision==0.26.0
[pip3] triton==3.6.0

cc @chauhang @penguinwu @voznesenskym @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @aakhundov @coconutruben @jataylo

**Comment by jansel** [comment]
My agent thinks this is intended:

Conclusion:
- Classified as INTENDED_BEHAVIOR. Inductor is producing a small float32 reduction-order difference before F.normalize; the user-visible large difference is caused by applying F.normalize with eps=1e-12 to an almost-zero vector, which is numerically ill-conditioned. No root-cause Inductor correctness bug was found and no source changes/tests were added.
